pkg/apiserver/resthandler.go

```diff
@@ -114,23 +121,83 @@ func getResourceHandler(scope RequestScope, getter getterFunc) restful.RouteFunc
 	}
 }
 
+func parseBool(str string) (bool, error) {
+	if len(str) == 0 {
+		return false, nil
+	}
+	return strconv.ParseBool(str)
+}
+
+func getExportOptions(req *restful.Request) (ExportOptions, error) {
+	var err error
+	result := ExportOptions{}
+	if result.Export, err = parseBool(req.QueryParameter("export")); err != nil {
+		return result, err
+	}
+	result.Exact, err = parseBool(req.QueryParameter("exact"))
+	return result, err
+}
+
+func exportObjectMeta(objMeta *api.ObjectMeta, exact bool) {
+	objMeta.UID = ""
+	if !exact {
+		objMeta.Namespace = ""
+	}
+	objMeta.CreationTimestamp = unversioned.Time{}
+	objMeta.DeletionTimestamp = nil
+	objMeta.ResourceVersion = ""
+	objMeta.SelfLink = ""
+	if len(objMeta.GenerateName) > 0 && !exact {
+		objMeta.Name = ""
+	}
+}
+
+func maybeExport(obj runtime.Object, req *restful.Request, scope RequestScope, e rest.Exporter) (runtime.Object, error) {
+	opts, err := getExportOptions(req)
+	if err != nil {
+		return nil, err
+	}
+	if opts.Export {
+		if e == nil {
+			return nil, errors.NewBadRequest(fmt.Sprintf("export unsupported for: %s", scope.Kind))
+		}
+		if meta, err := api.ObjectMetaFor(obj); err == nil {
```

Comment:
Why force every strategy to put the same boilerplate at the top of each export method?